server/ladder_service/ladder_service.py

```diff
@@ -458,55 +519,38 @@ def get_player_mean(player):
             # Really in the future, just send a better description
 
             self._logger.debug("Starting ladder game: %s", game)
-            # Options shared by all players
-            options = GameLaunchOptions(
-                mapname=mapname,
-                expected_players=len(all_players),
-                game_options=game_options
-            )
 
             def make_game_options(player: Player) -> GameLaunchOptions:
-                return options._replace(
+                return GameLaunchOptions(
+                    mapname=mapname,
+                    expected_players=len(all_players),
+                    game_options=game_options,
                     team=game.get_player_option(player.id, "Team"),
-                    faction=player.faction,
+                    faction=game.get_player_option(player.id, "Faction"),
```

Comment:
we are now setting this to `player.faction.value`. Is there a difference to plain `faction.value`?

Reply:
I don't think so, since Enums should get serialized to their value anyways when running them through `json.dumps`

Reply:
It might make it harder to tell what the faction is in the TRACE logs as before the faction would be printed as `repr` of the enum which includes both the name and value.